api/scpca_portal/models/dataset.py

```diff
@@ -342,35 +339,31 @@ def original_files(self) -> Iterable[OriginalFile]:
     def original_file_paths(self) -> Set[Path]:
         return {Path(of.s3_key) for of in self.original_files}
 
-    @property
-    def metadata_file_map(self) -> Dict[str, str]:
-        metadata_file_map = {}
+    def get_metadata_file_content(self, libraries: Iterable[Library]) -> str:
+        libraries_metadata = Library.get_libraries_metadata(libraries)
+        return metadata_file.get_file_contents(libraries_metadata)
+
+    def get_project_modality_libraries(
+        self, project_id: str, modality: Modalities
+    ) -> Iterable[Library]:
+        return Library.objects.filter(samples__in=self.get_samples(project_id, modality)).distinct()
+
+    def get_project_modality_metadata_file_content(self, project_id: str, modality: Modalities):
+        libraries = self.get_project_modality_libraries(project_id, modality)
+        return self.get_metadata_file_content(libraries)
+
+    def get_metadata_file_contents(self) -> List[tuple[str, Modalities, str]]:
```

Comment:
If dataset format is metadata (only), then we should return a list of one tuple where project and modality are `None`

```suggestion
    def get_metadata_file_contents(self) -> List[tuple[Union[str | None], Union[Modalities, None], str]]:
```